### ISSUE TITLE
[!!] Updated Filters in Step 4

### ISSUE THREAD

**Issue by bakkenb** [w3c/wcag-em-report-tool#317]
Link to proposed Step 4 page with updated filters:
http://w3c.github.io/wcag-em-report-tool/dist/#!/evaluation/audit

I played with the filters some more. Currently the filter titled "WCAG 2.1" shows only the 17 new Success Criteria, not all Success Criteria in 2.1. Although I see a strong use case for only wanting to see those Criteria, I find the button title inaccurate and possibly misleading to those who are new to accessibility.

I am okay with the functionality that if no filters are selected (WCAG version or conformance level) nothing shows. That is common sense.

Based on brief EOWG discussion and followup, I suggest:
1. Let users select whatever combination of filters they want.
2. If they have nothing selected in one set, display a message such as, "No success criteria are selected. Change the Filter selection above to list success criteria."
3. Change the functionality or change the label of "WCAG 2.1". Either:
3.a. Functionality: "WCAG 2.1" actually shows all of the WCAG 2.1 Success Criteria or,
3.b. Label is something like "New in WCAG 2.1"

I can think of a use case where people would want to do a report only on the new 2.1 Success Criteria, therefore I lean towards scenario 3.b but would like to hear what the group thinks.

**Comment by slhenry** [comment]
> Let users select whatever combination of filters they want.

+1

> If they have nothing selected in one set, display a message such as, "No success criteria are selected. Change the Filter selection above to list success criteria."

+1

> Change the functionality or change the label of "WCAG 2.1". Either:
>    3.a. Functionality: "WCAG 2.1" actually shows all of the WCAG 2.1 Success Criteria or,
 >   3.b. Label is something like "New in WCAG 2.1"

Based on:
> I see a strong use case for only wanting to ... do a report only on the new 2.1 Success Criteria

Not 3.a. Maybe 3.b. or another option. /me thinks...

**Comment by yatil** [comment]
In the quick reference we use 

- WCAG 2.1
- WCAG 2.0
- Only 2.1 Added Success Criteria

<img width="299" alt="Above, but in a screenshot" src="https://user-images.githubusercontent.com/58862/70390186-10406d00-19c9-11ea-953c-df94c818a6c0.png">

**Comment by nitedog** [comment]
@rvantonisse right now we have two buttons in Step 4, which can be selected and unselected in any combination. You have to know that you need to select "WCAG 2.1" and unselect "WCAG 2.0" to get the diff of criteria only in 2.1 (not really clear to figure out, even for advanced users). Also, you can unselect "WCAG 2.1" and "WCAG 2.0", which does not really make any sense.

Can we easily switch these to three buttons, of which exactly one of them is selected at any time - specifically: "WCAG 2.1" *or* "WCAG 2.0", *or* "New in WCAG 2.1" (in this order)?

**Comment by rvantonisse** [comment]
Thank you for the input!

> Let users select whatever combination of filters they want.

Yes.
Does this mean to leave the filter options unset, or can we keep initial set filters based on conformance target and WCAG version?

> If they have nothing selected in one set, display a message such as, "No success criteria are selected. Change the Filter selection above to list success criteria."

No and yes.
Looking at how the filters work currently and they actually work like an enabling button, not a filter. With no, or all filters checked I think the default should be to show all criteria, since no filters are applied. I agree that it is nice to have feedback for the user when no criteria is shown.

> Change the functionality or change the label of "WCAG 2.1"

I prefer something like option 3b. More like "Added since WCAG ...". 

And in combination with:

> You have to know that you need to select "WCAG 2.1" and unselect "WCAG 2.0" to get the diff of criteria only in 2.1 (not really clear to figure out, even for advanced users). Also, you can unselect "WCAG 2.1" and "WCAG 2.0", which does not really make any sense.
Can we easily switch these to three buttons, of which exactly one of them is selected at any time - specifically: "WCAG 2.1" or "WCAG 2.0", or "New in WCAG 2.1" (in this order)?

Adding three options for version filtering only makes sense to me when applied like in the quickref as a select, like yatil showed. I prefer to stay with 2 WCAG version filtering and rephrase it like mentioned by bakkenb in 3b.

**Comment by cochfarf** [comment]
I don't know if it just me but selecting WCAG 2.0 on Step 1 does not uncheck WCAG 2.1 in Step 4

**Comment by slhenry** [comment]
> Does this mean to leave the filter options unset, or can we keep initial set filters based on conformance target and WCAG version?

+1 to keep the initial selection in Step 4 matching what they selected in Step 1 -- and also let them change it in Step 4 if they want to.

> Looking at how the filters work currently and they actually work like an enabling button

Maybe instead of "Filter" it should be "Show" or something else??

>  I prefer to stay with 2 WCAG version filtering and rephrase it like mentioned by bakkenb in 3b.

In a different situation, I would agree -- e.g., if this was an in-house tool that users were trained on.

Since it's not, I don't think we can count on users figuring it out, especially without a lot of frustration.

**Comment by rvantonisse** [comment]
> I don't know if it just me but selecting WCAG 2.0 on Step 1 does not uncheck WCAG 2.1 in Step 4

Thank you for your comment @cochfarf .

Asume that it does (If it is not implemented now, it will be like how it is done with the level filtering). It will only set filter values before you visit step 4. After visiting step 4, step 1 does not anymore affect the filter in step 4. if you want to test this, make sure to refresh the page at step one, adjust version and or conformance target and open step 4.

**Comment by nitedog** [comment]
@rvantonisse pre-setting the filters seems to not work on some browsers? In any case, I think that is a separate issue (Issue #317)?

**Comment by nitedog** [comment]
@slhenry I don't understand what you are suggesting. Would the following work for you?

> Show: [New in WCAG 2.1] [WCAG 2.0]

Thaz is, change the label from "filters" to "show", and change the button name "WCAG 2.1" to "New in WCAG 2.1". If not, please explain what you find confusing or what alternative you suggest.

**Comment by daniel-montalvo** [comment]
I think having three checkboxes instead of the current two would be better.

1. We would have consistency with the quick reference as @yatil pointed out.
2. We would make clearer that you can indeed select only those new criteria added in WCAG 2.1. Yes, you can do it now but you have to know in advance. I agree that we should not expect everybody using this resource to be aware of that.

**Comment by bakkenb** [comment]
What Daniel mentions is what I was trying to say in the planning meeting on Wednesday. I feel it would be better too, but I understand that it is not actually needed. Would have to see the two buttons in place and working to know whether I feel a third would be better for those trying to use the resource for the first time, or those not familiar with the difference between 2.0 and 2.1.

**Comment by nitedog** [comment]
@bakkenb @slhenry @daniel-montalvo please check if the changes made by @rvantonisse address your concern: http://w3c.github.io/wcag-em-report-tool/dist/#!/evaluation/audit

**Comment by daniel-montalvo** [comment]
> 
> 
> @bakkenb @slhenry @daniel-montalvo please check if the changes made by @rvantonisse address your concern: http://w3c.github.io/wcag-em-report-tool/dist/#!/evaluation/audit

Yes, it does. It is clearer now. I find it helpful that the two checks are selected if you have previously chosen WCAG 2.1 in step 1. I think adding "New in 2.1" communicates that selecting 2.1 in step 1 means SC from 2.0 + new SC in 2.1, and we are showing these checked in step 4, which clarifies the purpose of the filters. In addition, we are letting them choose for a second time in step 4 what they want to evaluate as they could uncheck one of these filters.

I still feel consistency with how the quick reference presents the information would be worthy but definitely not feeling strongly.

**Comment by iadawn** [comment]
> I still feel consistency with how the quick reference presents the information would be worthy but definitely not feeling strongly.

+1 to this... although I feel a little bit more strongly. There is an element of thinking require along the links of 'to get 2.1 I have to click both 2.0 and new in 2.1'. This goes away if you have three options: 2.0, 2.1, Only 2.1.

**Comment by slhenry** [comment]
RESOLUTION from [EOWG telecon 13 Dec](https://www.w3.org/2019/12/13-eo-minutes.html#item01)

same looking buttons, with options/radio buttons (instead of checkboxes):

()WCAG 2.1     ()New in WCAG 2.1      ()WCAG 2.0

**Comment by yatil** [comment]
I’m mildly concerned with this resolution. (But not enough to warrant holding up publishing this resource.)

1. The order and wording is different to the Quickref. We should try to be consistent. (We can change the Quickref wording, I just want us to be consistent.)
2. New in WCAG 2.1 will get old once WCAG 2.2 is released. Unless we change the wording or make sure that we can release a new version with 2.2, front loading with New might be confusing. 
3. “Added in 2.1” seems more accurate as WCAG 2.1 is not exactly “new” anymore. And could be extended for other 2.x updates (assuming they don’t remove SCs or change their levels).

**Comment by slhenry** [comment]
For easy reference,
* the [updated Report Tool](http://w3c.github.io/wcag-em-report-tool/dist/#!/evaluation/audit) currently has:
>* WCAG 2.1
>* New in WCAG 2.1
>* WCAG 2.0

* the [QuickRef ](https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_customize) currently has:
> * WCAG 2.1
> * WCAG 2.0
> * Only 2.1 Added Success Criteria

I agree consistency is generally good. Although with differently audiences, we might want minor differences (explained below). Also, once people are used to an interface, (like the QuickRef), it's generally better not to change it unless strongly warranted.

The order in this Report Tool makes a bit more sense for my brain:
* WCAG 2.1
* New in WCAG 2.1
* WCAG 2.0
although my brain doesn't have any trouble processing the different order in the QuickRef.

I think "New" is OK when you're talking about criteria that are added, even after time. "New in 2.1" will always be correct, even years from now. i.e., it's new from 2.0 to 2.1, rather than it's new today.

In the future, we could have:
* WCAG 2.2
* New in WCAG 2.2
* WCAG 2.1
* New in WCAG 2.1
* WCAG 2.0

I'm also OK with "Added in 2.1" instead of "New in 2.1", i.e.,
* WCAG 2.2
* Added in WCAG 2.2
* WCAG 2.1
* Added in WCAG 2.1
* WCAG 2.0

If both work OK for others, then probably we want to use "Added in WCAG 2.1" so it's closer to the QuickRef.

(Note that this target audience should know (or easily figure out) that this si "success criteria" so I don't think we need to include that. However, the QuickRef audience has less knowledge. For that, we probably do want to include "Success Criteria".)

(related comment in @@)

**Comment by yatil** [comment]
+1 to “Added in WCAG 2.1”

**Comment by nitedog** [comment]
Updated (final) RESOLUTION from [EOWG telecon 20 Dec](https://www.w3.org/2019/12/20-eo-minutes.html#item01):

Change "New in WCAG 2.1" to "Added in WCAG 2.1"
